lib/utils/thingtalk/describe.ts

```diff
@@ -549,58 +569,226 @@ export class Describer {
         return recursiveHelper(expr);
     }
 
-    private _getDeviceAttribute(selector : Ast.DeviceSelector, name : string) : unknown|undefined {
+    private _getDeviceAttribute(selector : Ast.DeviceSelector, name : string) : InterpChunk|undefined {
         for (const attr of selector.attributes) {
             if (attr.name === name)
                 return this._describeArg(attr.value, {});
         }
         return undefined;
     }
 
+    private _computeParamMatchingScore(exampleInParams : Ast.InputParam[],
+                                       programInParams : Ast.InputParam[],
+                                       exampleArgs : Record<string, Type>) {
```

Comment:
It's minor, but there are a few cases the function type is not annotated...

Reply:
We don't require type annotations on function return types when it's inferrable, do we? Otherwise there are a lot of places where we need to add types.